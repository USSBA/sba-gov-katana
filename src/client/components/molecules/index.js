import Breadcrumb from "./breadcrumb/breadcrumb.jsx";
import ButtonCta from "./button-cta/button-cta.jsx";
import CallToAction from "./call-to-action/call-to-action.jsx";
import Callout from "./callout/callout.jsx";
import Card from "./card/card.jsx";
import CardGrid from "./card-grid/card-grid.jsx";
import ContactCardLookup from "./contact-card-lookup/contact-card-lookup.jsx";
import DocumentArticle from "./document-article/document-article.jsx";
import DocumentCard from "./document-card/document-card.jsx";
import DocumentLookup from "./document-lookup/document-lookup.jsx";
import DropdownMenu from "./dropdown-menu/dropdown-menu.jsx";
import FeaturedCallout from "./featured-callout/featured-callout.jsx";
import FeedbackForm from "./feedback-form/feedback-form.jsx";
import FormPageButtons from "./form-page-buttons/form-page-buttons.jsx";
import GoogleTranslate from "./google-translate/google-translate.jsx";
import ImageSection from "./image-section/image-section.jsx";
import Lookup from "./lookup/lookup.jsx";
import PageLinkGroup from "./page-link-group/page-link-group.jsx";
import Paginator from "./paginator/paginator.jsx";
import ParagraphPlaceholder from "./paragraph-placeholder/paragraph-placeholder.jsx";
import PreviousNextSection from "./previous-next/previous-next.jsx";
import QuickLinks from "./quick-links/quick-links.jsx";
import ReadMore from "./readmore-section/readmore-section.jsx";
import ResourceCard from "./resource-card/resource-card.jsx";
import SbicLookup from "./sbic-lookup/sbic-lookup.jsx";
import SearchBar from "./search-bar/search-bar.jsx";
import SectionHeader from "./section-header/section-header.jsx";
import SbaModal from "./sba-modal/sba-modal.jsx"
import SimpleCta from "./simple-cta/simple-cta.jsx";
import SmallInverseCta from "./small-inverse-cta/small-inverse-cta.jsx";
import StyleGrayBackground from "./style-gray-background/style-gray-background.jsx";
import SubMenu from "./sub-menu/sub-menu.jsx";
import SubsectionHeader from "./subsection-header/subsection-header.jsx";
import SuretyLookup from "./surety-lookup/surety-lookup.jsx";
import TenStepsNav from "./ten-steps-nav/ten-steps-nav.jsx";
import TenStepsSection from "./ten-steps-section/ten-steps-section.jsx";
import TextReadMoreSection from "./text-readmore-section/text-readmore-section.jsx";
import TileCollection from "./tile-collection/tile-collection.jsx";
import TitleBox from "./title-box/title-box.jsx";
import TitleSection from "./title-section/title-section.jsx";
import TextSection from "./text-section/text-section.jsx";

export { Breadcrumb, ButtonCta, CallToAction, Callout, Card, CardGrid, ContactCardLookup, DocumentArticle, DocumentCard, DocumentLookup, DropdownMenu, FeaturedCallout, FeedbackForm, FormPageButtons, GoogleTranslate, ImageSection, Lookup, PageLinkGroup, Paginator, ParagraphPlaceholder, PreviousNextSection, QuickLinks, ReadMore, ResourceCard, SbicLookup, SearchBar, SectionHeader, SimpleCta, SmallInverseCta, StyleGrayBackground, SubMenu, SubsectionHeader, SuretyLookup, TenStepsNav, TenStepsSection, TextReadMoreSection, TextSection, TileCollection, TitleBox, TitleSection, SbaModal };
