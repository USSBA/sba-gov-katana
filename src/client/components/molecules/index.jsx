import Address from './address/address'
import AuthorCard from './author-card/author-card'
import BizzieChatbotWidget from './bizzie-chatbot-widget/bizzie-chatbot-widget'
import Breadcrumb from './breadcrumb/breadcrumb'
import ButtonCta from './button-cta/button-cta'
import CallToAction from './call-to-action/call-to-action'
import Card from './card/card'
import CardGrid from './card-grid/card-grid'
import ContactCard from './contact-card/contact-card'
import ContactCardLookup from './contact-card-lookup/contact-card-lookup'
import DetailCard from './detail-card/detail-card'
import DistrictOfficePreviousNextSection from './district-office-previous-next/district-office-previous-next'
import FeedbackForm from './feedback-form/feedback-form'
import FormPageButtons from './form-page-buttons/form-page-buttons'
import GoogleTranslate from './google-translate/google-translate'
import LenderResultCard from './lender-result-card/lender-result-card'
import MainLoader from './main-loader/main-loader'
import MenuTile from './menu-tile/menu-tile'
import MenuTileWithLinks from './menu-tile-with-links/menu-tile-with-links'
import NewsletterForm from './newsletter-form/newsletter-form'
import NoResultsSection from './no-results-section/no-results-section'
import NumberedCalloutBlock from './numbered-callout-block/numbered-callout-block'
import PageLinkGroup from './page-link-group/page-link-group'
import Paginator from './paginator/paginator'
import PhoneNumber from './phone-number/phone-number'
import PreviousNextSection from './previous-next/previous-next'
import QuickLinks from './quick-links/quick-links'
import ReadMore from './readmore/readmore'
import ReadMoreSection from './readmore-section/readmore-section'
import RemoveMainLoader from './main-loader/remove-main-loader'
import SbaModal from './sba-modal/sba-modal'
import SbaNewsModal from './news-modal/news-modal'
import SearchBar from './search-bar/search-bar'
import SuggestedRouteCard from './suggested-route-card/suggested-route-card'
import LongScrollNav from './long-scroll-nav/long-scroll-nav'
import TitleSection from './title-section/title-section'
import Typeahead from './typeahead/typeahead'

export {
  Address,
  AuthorCard,
  BizzieChatbotWidget,
  Breadcrumb,
  ButtonCta,
  CallToAction,
  Card,
  CardGrid,
  ContactCard,
  ContactCardLookup,
  DetailCard,
  DistrictOfficePreviousNextSection,
  FeedbackForm,
  FormPageButtons,
  GoogleTranslate,
  LenderResultCard,
  MainLoader,
  MenuTile,
  MenuTileWithLinks,
  NewsletterForm,
  NoResultsSection,
  NumberedCalloutBlock,
  PageLinkGroup,
  Paginator,
  PhoneNumber,
  PreviousNextSection,
  QuickLinks,
  ReadMore,
  ReadMoreSection,
  RemoveMainLoader,
  SbaModal,
  SbaNewsModal,
  SearchBar,
  SuggestedRouteCard,
  LongScrollNav,
  TitleSection,
  Typeahead
}
