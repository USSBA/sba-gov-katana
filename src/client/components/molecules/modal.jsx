import React from "react";
import Modal from 'react-modal'
import styles from './modal.scss'
import sbaLogo from '../../../../public/assets/images/logo.png';
import exitIcon from '../../../../public/assets/svg/exit-modal-close.svg';

class SbaModal extends React.Component {
  render() {
    let logo = this.props.showLogo ? (<img className={styles.logo} src={sbaLogo}/>) : undefined;
    return (
      <Modal isOpen={true} className={styles.content} overlayClassName={styles.overlay}>
        {logo}
        <a onClick={this.props.onClose}><img className={styles.exitIcon} src={exitIcon}/></a>
        <h2 className={styles.title}>{this.props.title}</h2>
        <div className={styles.divider}></div>
        <p className={styles.text}>{this.props.text}</p>
        {this.props.children}
        <div className={styles.btnContainer}>
          <button onClick={this.props.onClose} className={styles.btnCancel}>{this.props.cancelButtonText}</button>
          <button onClick={this.props.onClickOk} className={styles.btnContinue}>{this.props.okButtonText}</button>
        </div>
      </Modal>
    );
  }
}

SbaModal.propTypes = {
  cancelButtonText: React.PropTypes.string,
  okButtonText: React.PropTypes.string,
  showCancel: React.PropTypes.boolean,
  showOk: React.PropTypes.boolean,
  title: React.PropTypes.string.isRequired,
  text: React.PropTypes.string.isRequired,
  showLogo: React.PropTypes.boolean,
  onClickOk: React.PropTypes.func,
  onClose: React.PropTypes.func
}

SbaModal.defaultProps ={
  cancelButtonText: "Cancel",
  okButtonText: "OK",
  showCancel: true,
  showOk: true,
  title: "Sample Modal Title",
  text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat",
  showLogo: false,
  onClickOk: function() {
    console.log("Clicked OK")
  }
}

export default SbaModal;
