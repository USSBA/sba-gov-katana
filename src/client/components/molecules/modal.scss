.overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0,0,0,0.66);
}

.content {
    background-color: white;
    position: absolute;
    width: 665px;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
        -ms-transform: translate(-50%, -50%);
        -webkit-transform: translate(-50%, -50%);
        -moz-transform: translate(-50%, -50%);
        -o-transform: translate(-50%, -50%);
        transform: translate(-50%, -50%);
    margin: auto;
    padding: 40px 40px 30px;
    font-family: "Source Sans Pro";

    @media screen and (max-width: 767px) {
        width: 90vw;
        position: absolute;
        height: auto;
        margin: auto;
        padding: 8vw;
        left: 50%;
        top: 50%;
        -ms-transform: translate(-50%, -50%);
        -webkit-transform: translate(-50%, -50%);
        -moz-transform: translate(-50%, -50%);
        -o-transform: translate(-50%, -50%);
        transform: translate(-50%, -50%);
    }
}

.logo {
    display: block;
    margin: auto;
    margin-bottom: 25px;

    @media screen and (max-width: 767px) {
        width: 26.7vw;
        margin-bottom: 6.7vw;
        height: auto;
    }
}

.closeIcon {
    position: absolute;
    top:30px;
    right:30px;
    cursor: pointer;
}

.title {
    margin-bottom: 22px;
    text-align: center;

}

.divider {
    width: 160px;
    border: 2px solid #0B97DD;
    margin: auto;
    margin-bottom: 22px;

    @media screen and (max-width: 767px) {
        width: 100px;
    }
}

.text {
    text-align: center;
    margin-bottom: 30px;

}

.btnContainer {
    text-align: center;

    button {
        margin-right: 30px;
    }
}

// .btn {
//     width: 124px;
//     font-size: 18px;
//     font-weight: 900;
//     padding: 12px 0 12px;

//     @media screen and (max-width: 767px) {
//         width: 100%;
//     }
// }

// .btnCancel {
//     // composes: btn;
//     // color: #666666;
//     // background-color: white;
//     // border: 2px solid #666666;
//     margin-right: 30px;
//   //   &:hover,
//   //   &:focus {
//   //   // background-color: #666666;
//   //   // color: #fff;
//   // }
//     @media screen and (max-width: 767px) {
//         margin-right: 0;
//     }
// }

// .btnContinue {
//     composes: btn;
//     color: white;
//     background-color: #0093DF;
//     border: none;
//     position: relative;
//         @media screen and (max-width: 767px) {
//         margin-top: 20px;
//     }
// }
