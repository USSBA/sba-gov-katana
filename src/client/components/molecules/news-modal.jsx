import React from "react";
import ReactModal from "react-modal";
import { connect } from "react-redux";
import {bindActionCreators} from "redux";
import * as ModalActions from "../../actions/show-modal.js";
import * as ContentActions from "../../actions/content.js";
import styles from "./news-modal.scss";
import TextInput from "../atoms/text-input.jsx";
import exitIcon from "../../../../public/assets/svg/close_button.svg";
import {logEvent} from "../../services/analytics.js";
import constants from "../../services/constants.js";
import config from "../../services/client-config.js";
import {getEmailValidationState, getZipcodeValidationState, containsErrorOrNull} from "../../services/page-validator-helpers.js";
import {includes} from "lodash";
import envelopeIcon from '../../../../public/assets/svg/envelope.svg';
import SmallPrimaryButton from "../atoms/small-primary-button/small-primary-button.jsx";

class SbaNewsModal extends React.Component {
    timerId = null;

    constructor(props) {
        super();
        this.state = {
            userEmailAddress: "",
            userZipCode:  "",
            modalIsOpen: config.govdelivery,
            displayForm: true,
            validStates: {
                userEmailAddress: null,
                userZipCode: null
            }
        };
    }

    componentDidMount() {
        this.validateFields(["userEmailAddress", "userZipCode"]);
        window.scrollTo(0, 0);
        console.log(config.govdelivery);
    }


    validateSingleField(validationFunction, name, defaultWhenNotSuccessful) {
        let validationState = validationFunction(name, this.state[name], defaultWhenNotSuccessful || null);
        if(validationState[name] === "error"){
            logEvent({"category": "Newsletter Modal", "action": "Error Event", "label": name});
        }
        return validationState;
    }

    validateFields(fields, defaultWhenNotSuccessful) {
        let validStates = this.state.validStates;

        if (includes(fields, "userEmailAddress")) {
            validStates = Object.assign(validStates, this.validateSingleField(getEmailValidationState, "userEmailAddress", defaultWhenNotSuccessful));
        }
        if (includes(fields, "userZipCode")) {
            validStates = Object.assign(validStates, this.validateSingleField(getZipcodeValidationState, "userZipCode", defaultWhenNotSuccessful));
        }
        this.setState({validStates: validStates})
    }

    isValidForm() {
        return !containsErrorOrNull(this.state.validStates);
    }

    handleSubmit(e) {
        e.preventDefault();
        this.props.actions.fetchContentIfNeeded("newsletterRegistration", "newsletter-registration", {
            userEmailAddress: this.state.userEmailAddress,
            userZipCode: this.state.userZipCode
        });

        this.setState({displayForm: false});
        this.timerId = setTimeout(()=>{
            this.setState({modalIsOpen: false});
        }, 5000);
    }

    componentWillUnmount(){
        if(this.timerId != null){
            clearTimeout(timerId);
        }
    }

    handleChange(e) {
        let newState = {};
        let name = e.target.name;
        newState[name] = e.target.value;
        this.setState(newState, () => this.validateFields([name]));
    }

    handleBlur(e) {
        this.validateFields([e.target.name], "error");
    }

    handleFocus(nameOrEvent) {
        let name = nameOrEvent && nameOrEvent.target && nameOrEvent.target.name
            ? nameOrEvent.target.name
            : nameOrEvent;
        logEvent({"category": "Newsletter Modal", "action": "Focus Event", "label": name});
    }

    handleClose(e){
        this.setState({modalIsOpen: false});
    }

    render() {
        return (
            <ReactModal isOpen={this.state.modalIsOpen} className={styles.content} overlayClassName={styles.overlay} contentLabel="Modal">
                <div>
                    <h2 className={this.state.displayForm ? styles.title : styles.hideForm}>Get business advice and invitations to local business events.</h2>
                    <a onClick={this.handleClose.bind(this)}><img className={styles.exitIcon} src={exitIcon}/></a>
                </div>
                <div className={this.state.displayForm ? styles.showForm : styles.removeForm}>
                    <form id="newsletter-modal-form" ref={(input) => this.newsletterModalForm = input} onSubmit={(e) => this.handleSubmit(e)} >
                        <TextInput name="userEmailAddress" errorText={constants.messages.validation.invalidNewsLetterEmail} placeholder="Your email address"  handleChange={this.handleChange.bind(this)} value={this.state.userEmailAddress} getValidationState={this.state.validStates.userEmailAddress} onBlur={this.handleBlur.bind(this)} onFocus={this.handleFocus.bind(this)}/>
                        <div className="sba-blue">
                            <div className={styles.zipTextBox}><TextInput name="userZipCode" errorText={constants.messages.validation.invalidNewsLetterZipCode}  placeholder="Zip code"  handleChange={this.handleChange.bind(this)} value={this.state.userZipCode} getValidationState={this.state.validStates.userZipCode} maxLength="5" onBlur={this.handleBlur.bind(this)} onFocus={this.handleFocus.bind(this)}/></div>
                                <div className={styles.btnContainer}>
                                    <SmallPrimaryButton text="SUBSCRIBE" type="submit" disabled={!(this.isValidForm())}/>
                                </div>
                        </div>
                        <p className={styles.privacyLink}><a  href="about-sba/sba-performance/open-government/about-sbagov-website/privacy-policy">Privacy policy</a></p>
                    </form>
                </div>

                <div className={this.state.displayForm ? styles.hideForm : styles.showForm}>
                    <img className={styles.envelopeIcon} src={envelopeIcon}/>
                    <p className={styles.byeMsg}>Thanks for subscribing!</p>
                </div>
            </ReactModal>
        );
    }
}

function mapStateToProps(state) {
    return {
        userEmailAddress: state.modalReducer.modalProps.userEmailAddress
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(ContentActions, dispatch),
        modalActions: bindActionCreators(ModalActions, dispatch)
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(SbaNewsModal);