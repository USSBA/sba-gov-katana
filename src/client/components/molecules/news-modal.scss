
.overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 20;
  background-color: rgba(0,0,0,0.66);
}

.content {
  background-color: #D6ECFC;
  position: relative;
  width: 400px;
  height: auto;
  top: 50%;
  transform: translate(5%,-50%);
  margin: auto;
  padding: 20px;
  font-family: "Source Sans Pro";
}

.exitIcon {
  position: absolute;
  top:30px;
  right:30px;
  width: 26.16px;
  height: 26.16px;
  color: #666666;
  transform: scaleX(-1);
  border-radius: 1px;
  cursor: pointer;
}
.removeForm{
  display: none;
}

.hideForm{
  visibility: hidden;

}
.showForm{
  visibility: visible;
}

.envelopeIcon{
  position: absolute;
  height: 33px;
  width: 54px;
  top: 120px;
  right: 170px;
}

.byeMsg{
  position: absolute;
  height: 26px;
  width: 268px;
  top: 160px;
  right: 50px;
  color: #333333;
  font-size: 24px;
  font-weight: 900;
  line-height: 26px;
}

.title {
  height: 44px;
  width: 280px;
  color: #333333;
  font-family: "Source Sans Pro";
  margin-bottom: 30px;
  font-size: 18px;
  font-weight: bold;
  line-height: 22px;
}

.zipTextBox {
  display: inline-block;
  height: 0px;
  width: 180px;
  background-color: #FFFFFF;
}

.btnContainer{
  display: inline-block;
  width: 50%;
  padding-left: 20px;
}

.privacyLink{
  margin-top: 30px;
  text-align: center;
  vertical-align: top;
}

