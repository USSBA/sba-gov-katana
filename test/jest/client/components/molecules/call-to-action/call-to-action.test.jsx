import React from 'react'
import { CallToAction } from 'molecules'

import renderer from 'react-test-renderer'
import _ from 'lodash'

import { render, cleanup } from 'react-testing-library'

import 'jest-dom/extend-expect'

afterEach(cleanup)

const ctaProps = {
  size: 'Small',
  buttonAction: {
    buttonText: 'Click Here',
    file: 'document.txt',
    type: 'file'
  }
}

describe('CallToAction', () => {
  it('renders an image on the CTA when passed in image props for prop size of "Large"', async () => {
    ctaProps.image = { url: '/images/photo.png' }
    ctaProps.size = 'Large'
    const { getByTestId } = render(<CallToAction {...ctaProps} />)
    const ctaImage = getByTestId('image')
    expect(ctaImage).toBeInTheDocument()
  })

  it('renders a headline passed in as props', async () => {
    ctaProps.headline = 'We are Fearless'
    const { getByTestId } = render(<CallToAction {...ctaProps} />)
    const ctaHeadline = getByTestId('headline')
    expect(ctaHeadline).toBeInTheDocument()
    expect(ctaHeadline).toHaveTextContent(ctaProps.headline)
  })

  it('renders a blurb passed in as props for prop size of "Medium"', async () => {
    ctaProps.blurb = 'Come join our herd'
    ctaProps.size = 'Medium'
    const { getByTestId } = render(<CallToAction {...ctaProps} />)
    const ctaBlurb = getByTestId('blurb')
    expect(ctaBlurb).toBeInTheDocument()
    expect(ctaBlurb).toHaveTextContent(ctaProps.blurb)
  })

  it('renders a blurb passed in as props for prop size of "Large"', async () => {
    ctaProps.blurb = 'Come join our herd'
    ctaProps.size = 'Large'
    const { getByTestId } = render(<CallToAction {...ctaProps} />)
    const ctaBlurb = getByTestId('blurb')
    expect(ctaBlurb).toBeInTheDocument()
    expect(ctaBlurb).toHaveTextContent(ctaProps.blurb)
  })

  it('renders a button with title text passed as props', async () => {
    ctaProps.buttonAction = {
      link: {
        title: 'Search Nearby',
        url: '/local-assistance/find'
      },
      type: 'link'
    }
    const { getByTestId } = render(<CallToAction {...ctaProps} />)
    const ctaButton = getByTestId('button')
    expect(ctaButton).toBeInTheDocument()
    expect(ctaButton).toHaveTextContent(ctaProps.buttonAction.link.title)
  })
})

describe('CallToAction snapshot tests', () => {
  const ctaOptions = {
    blurb: 'This is the blurb for the CTA.',
    buttonAction: {
      id: 9262,
      type: 'link',
      link: {
        url: 'http://www.example.com/',
        title: 'Call To Action'
      }
    },
    headline: 'This is the headline for the CTA.',
    image: {
      url: 'http://example.com/someImage.jpg',
      alt: 'alt text for image',
      width: 1000,
      height: 667
    },
    type: 'callToAction',
    title: 'National Small Business Week CTA',
    id: 9264,
    updated: 1519332130,
    created: 1519331979,
    langCode: 'en',
    style: 'Medium'
  }
  test('renders a small size', () => {
    const component = renderer.create(<CallToAction size="Small" {...ctaOptions} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('renders a medium size', () => {
    const component = renderer.create(<CallToAction size="Medium" {...ctaOptions} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('renders a large size', () => {
    const component = renderer.create(<CallToAction size="Large" {...ctaOptions} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('renders a button only', () => {
    const component = renderer.create(<CallToAction size="Button only" {...ctaOptions} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
