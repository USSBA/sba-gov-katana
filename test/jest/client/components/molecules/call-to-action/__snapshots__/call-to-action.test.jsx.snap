// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction renders a button only 1`] = `
<div
  className="btnOnly"
>
  <div
    className="ctaContainer"
    id="call-to-action"
  >
    <div
      className="image"
    >
      <div
        style={
          Object {
            "background": "url(http://example.com/someImage.jpg) no-repeat center center",
            "backgroundSize": "cover",
            "height": "100%",
            "width": "100%",
          }
        }
        title="alt text for image"
      />
    </div>
    <div
      className="contentContainer"
    >
      <h4
        className="headline"
      >
        This is the headline for the CTA.
      </h4>
      <p
        className="blurb"
      >
        This is the blurb for the CTA.
      </p>
      <a
        className="alternate primary link spacing"
        onClick={[Function]}
        style={Object {}}
      >
        Call To Action
      </a>
    </div>
    <img
      alt=""
      className="cornerGraphicLarge"
      src="test-file-stub"
    />
    <img
      alt=""
      className="cornerGraphicSmall"
      src="test-file-stub"
    />
  </div>
</div>
`;

exports[`CallToAction renders a large size 1`] = `
<div
  className="large"
>
  <div
    className="ctaContainer"
    id="call-to-action"
  >
    <div
      className="image"
    >
      <div
        style={
          Object {
            "background": "url(http://example.com/someImage.jpg) no-repeat center center",
            "backgroundSize": "cover",
            "height": "100%",
            "width": "100%",
          }
        }
        title="alt text for image"
      />
    </div>
    <div
      className="contentContainer"
    >
      <h4
        className="headline"
      >
        This is the headline for the CTA.
      </h4>
      <p
        className="blurb"
      >
        This is the blurb for the CTA.
      </p>
      <a
        className="alternate primary link spacing"
        onClick={[Function]}
        style={Object {}}
      >
        Call To Action
      </a>
    </div>
    <img
      alt=""
      className="cornerGraphicLarge"
      src="test-file-stub"
    />
    <img
      alt=""
      className="cornerGraphicSmall"
      src="test-file-stub"
    />
  </div>
</div>
`;

exports[`CallToAction renders a medium size 1`] = `
<div
  className="medium"
>
  <div
    className="ctaContainer"
    id="call-to-action"
  >
    <div
      className="image"
    >
      <div
        style={
          Object {
            "background": "url(http://example.com/someImage.jpg) no-repeat center center",
            "backgroundSize": "cover",
            "height": "100%",
            "width": "100%",
          }
        }
        title="alt text for image"
      />
    </div>
    <div
      className="contentContainer"
    >
      <h4
        className="headline"
      >
        This is the headline for the CTA.
      </h4>
      <p
        className="blurb"
      >
        This is the blurb for the CTA.
      </p>
      <a
        className="alternate primary link spacing"
        onClick={[Function]}
        style={Object {}}
      >
        Call To Action
      </a>
    </div>
    <img
      alt=""
      className="cornerGraphicLarge"
      src="test-file-stub"
    />
    <img
      alt=""
      className="cornerGraphicSmall"
      src="test-file-stub"
    />
  </div>
</div>
`;

exports[`CallToAction renders a small size 1`] = `
<div
  className="small"
>
  <div
    className="ctaContainer"
    id="call-to-action"
  >
    <div
      className="image"
    >
      <div
        style={
          Object {
            "background": "url(http://example.com/someImage.jpg) no-repeat center center",
            "backgroundSize": "cover",
            "height": "100%",
            "width": "100%",
          }
        }
        title="alt text for image"
      />
    </div>
    <div
      className="contentContainer"
    >
      <h4
        className="headline"
      >
        This is the headline for the CTA.
      </h4>
      <p
        className="blurb"
      >
        This is the blurb for the CTA.
      </p>
      <a
        className="alternate primary link spacing"
        onClick={[Function]}
        style={Object {}}
      >
        Call To Action
      </a>
    </div>
    <img
      alt=""
      className="cornerGraphicLarge"
      src="test-file-stub"
    />
    <img
      alt=""
      className="cornerGraphicSmall"
      src="test-file-stub"
    />
  </div>
</div>
`;
